examples/deploy/terraform/cluster/variables.tf

```diff
@@ -82,3 +82,27 @@ variable "kms_info" {
   })
   default = null
 }
+
+variable "irsa_policies" {
+  description = "Mappings for custom IRSA configurations."
+  type = list(object({
+    name                = string
+    namespace           = string
+    serviceaccount_name = string
+    policy              = string #json
+  }))
+
+  default = []
+}
+
+variable "irsa_external_dns" {
+  description = "Mappings for custom IRSA configurations."
+  type = object({
+    enabled             = optional(bool, false)
```

Comment:
why have this if you could make irsa_external_dns nillable instead? it would also simplify all the checking logic.

Reply:
conditional on nillables cause issues, i.e bastion